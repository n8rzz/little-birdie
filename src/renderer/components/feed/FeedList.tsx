import * as React from 'react';
import { FeedItem } from './FeedItem';

interface IProps {
  items: any;
}

export const FeedList: React.FC<IProps> = (props: IProps) => {
  if (props.items.length === 0) {
    return (
      <section>
        <div>No Feeds</div>
      </section>
    );
  }

  return (
    <section>
      <ul>
        {props.items.map((feed: any, index: number) => (
          <FeedItem
            key={`feedItem-${index}`}
            name={''}
            url={''}
          />
        ))}
      </ul>
    </section>
  );
};
