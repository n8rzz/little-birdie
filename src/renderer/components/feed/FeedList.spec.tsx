import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { FeedList } from './FeedList';

describe('FeedList', () => {
  const props = {
    items: ['Github', 'npm', 'Digital Ocean'],
  };

  it('should match snapshot', () => {
    const tree = TestRenderer.create(<FeedList items={props.items} />);

    expect(tree).toMatchSnapshot();
  });

  describe('when passed an empty #feedList', () => {
    it('should render default text', () => {
      const { getByText } = render(<FeedList items={[]} />);

      expect(getByText('No Feeds')).not.toBeUndefined();
    });
  });
});
