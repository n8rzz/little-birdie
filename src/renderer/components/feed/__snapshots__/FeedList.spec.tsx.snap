// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedList should match snapshot 1`] = `
<section>
  <ul>
    <li
      className="providerList-item"
    >
      <a
        href="#"
      >
        heroku
      </a>
    </li>
    <li
      className="providerList-item"
    >
      <a
        href="#"
      >
        heroku
      </a>
    </li>
    <li
      className="providerList-item"
    >
      <a
        href="#"
      >
        heroku
      </a>
    </li>
  </ul>
</section>
`;
